### PR TITLE
feat(bots): Leitura da Mesa única (issue #234 — slice 1)

### PR DESCRIPTION
**Pull request by Dhinihan**
## Resumo

Slice 1 da #234: `lerMesa` como snapshot único para linha fria e quente.

## Revisões endereçadas

| Fonte | Ajuste |
|-------|--------|
| Codex P2 | Mão e líder avaliados em lotes separados (`cartasPorJogador` correto) |
| CodeRabbit | `cartasQueNaoFazemVaza` reutilizado; flags da leitura em vencedora segura |
| @Dhinihan | `estado` removido de `podeAtravessar` / `escolherTravessia` e vencedora segura |
| @Dhinihan | `mesaTemQuemNaoQuerVaza` na `LeituraDaMesa`; `podeBifurcar` só lê a leitura |
| @Dhinihan (nit) | Testes de `liderQuerVaza`, `empates` e `mesaTemQuemNaoQuerVaza` |

739 testes passando.

<div><a href="https://cursor.com/agents/bc-552fdb1a-5f9e-45b9-8f8b-c3a399c0e9f0"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-552fdb1a-5f9e-45b9-8f8b-c3a399c0e9f0"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

## Summary by CodeRabbit

* **Refactor**
  * Bot decision logic now uses a single, shared board evaluation so openings, mid-round plays and endgame choices are more consistent and coherent.
* **Bug Fixes / Reliability**
  * Improved handling of edge cases (empty hand, leader-less table) leading to more predictable bot moves.
* **Tests**
  * Test suite updated to align with the new evaluation flow, increasing confidence in bot behavior.

[Review Change Stack](https://app.coderabbit.ai/change-stack/Dhinihan/fdp-online/pull/235?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: src/adapters/bots/pode-atravessar.ts [hunk 1]
```diff
@@ -1,36 +1,35 @@
...
-import { liderQuerVaza, urgenciaAlta } from './contexto-posicao-mesa';
-import type { ContextoJogadaQuente } from './contextoLinhaQuente';
+import type { LeituraDaMesa } from './ler-mesa';
 import { ehAltaOuMelhor } from './predicados-carta-avaliada';
 import type { DecisaoCartaQuente } from './regras-linha-quente';
 import { cartaMaisBarata } from './selecao-por-score';
 
-export function podeAtravessar(estado: EstadoEmJogo, contexto: ContextoJogadaQuente): boolean {
+export function podeAtravessar(estado: EstadoEmJogo, leitura: LeituraDaMesa): boolean {
+  void estado;
```
Comment (Dhinihan) [inline src/adapters/bots/pode-atravessar.ts]: **Parâmetro `estado` morto.** Agora que tudo vem da `leitura`, `estado` virou vestigial aqui — o `void estado;` só existe pra calar o `no-unused-vars`. Sugiro **remover o parâmetro** de `podeAtravessar` e `escolherTravessia` e ajustar a chamada em `DecisorJogadaLinhaQuente`. Mesmo caso em `pode-vencedora-segura.ts:10`.

**Review by Dhinihan — COMMENTED** [review]
Revisão do Slice 1. Lint limpo e 736 testes passando localmente. Implementação fiel ao design e ao comportamento pré-refactor — `ContextoJogadaQuente`/`ContextoUltimoLinhaQuente`/`ContextoGuardaPosicao` removidos e todos os ramos consomem `LeituraDaMesa`. Boa decisão avaliar mão e líder em lotes separados (evita inflar `cartasPorJogador`). Dois ajustes menores abaixo, nenhum bloqueante.

File: src/adapters/bots/pode-vencedora-segura.ts [hunk 2]
```diff
@@ -1,42 +1,39 @@
...
-import type { ContextoJogadaQuente } from './contextoLinhaQuente';
 import { existeGarantidaParaDepois } from './garantida-para-depois';
+import type { LeituraDaMesa } from './ler-mesa';
 import { ehAltaOuMelhor } from './predicados-carta-avaliada';
 import type { DecisaoCartaQuente } from './regras-linha-quente';
 import { cartaMaisBarata } from './selecao-por-score';
 
-export function podeTentarComVencedoraSegura(estado: EstadoEmJogo, contexto: ContextoJogadaQuente): boolean {
+export function podeTentarComVencedoraSegura(estado: EstadoEmJogo, leitura: LeituraDaMesa): boolean {
+  void estado;
```
Comment (Dhinihan) [inline src/adapters/bots/pode-vencedora-segura.ts]: **Parâmetro `estado` morto** (mesmo caso de `pode-atravessar.ts:9`). Remover `estado` de `podeTentarComVencedoraSegura` e `escolherVencedoraSegura`; o `void estado;` é só workaround do linter. `escolherEsperarOportunidade` pode manter o `estado` porque usa `estado.manilha` de verdade.

File: src/adapters/bots/regras-linha-quente.ts [hunk 3]
```diff
@@ -16,72 +17,59 @@ export interface DecisaoCartaQuente {
...
   });
 }
 
-function mesaPodePunir(estado: EstadoEmJogo, contexto: ContextoJogadaQuente, liderBaixa: number): boolean {
+function mesaPodePunir(estado: EstadoEmJogo, leitura: LeituraDaMesa, liderBaixa: number): boolean {
   return Boolean(
-    contexto.lider &&
-    contexto.lider.score <= liderBaixa &&
-    estado.mesa.some((item) => jogadorNaoQuerVaza(estado, item)),
+    leitura.lider && leitura.lider.score <= liderBaixa && estado.mesa.some((item) => jogadorNaoQuerVaza(estado, item)),
```
Comment (Dhinihan) [inline src/adapters/bots/regras-linha-quente.ts]: **Snapshot incompleto.** Combinamos no design "snapshot completo, ramos leem só campos", mas `mesaPodePunir` ainda alcança `estado.mesa.some(...)` pra saber se há jogador que já cumpriu na mesa. O `temAlvo` entrou na `LeituraDaMesa`, mas esse fato ficou de fora — enquanto ele lê `estado`, `podeBifurcar` não dá pra testar com uma fixture pura de `LeituraDaMesa`. Sugiro um campo na leitura (ex.: `mesaTemQuemNaoQuerVaza: boolean`). Como o Slice 2 reescreve este arquivo no módulo deep, dá pra resolver lá — mas é barato fazer já.

File: tests/adapters/ler-mesa.test.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      mesa: [{ jogadorId: 'j1', carta: criarCarta('K', '♣') }],
+      manilha: '5',
+    });
+    const leitura = lerMesa(estado, maoBasica);
+
+    expect(leitura.avaliadas).toHaveLength(maoBasica.length);
+    expect(leitura.lider?.carta).toEqual(criarCarta('K', '♣'));
+    expect(leitura.jogadorId).toBe(estado.maos[estado.jogadorAtual].jogador.id);
+  });
+});
```
Comment (Dhinihan) [inline tests/adapters/ler-mesa.test.ts:115]: _Nit opcional:_ os casos não cobrem `liderQuerVaza === true` nem classificação de `empates` (sempre 0 aqui). Lacuna pequena, vale um caso com empate na mão e um com líder ainda precisando fazer.
